jeremy2918/tsseract
# Added followers and following to the users model

# [Follows](https://trello.com/c/gbXxtAPJ/38-i-want-to-add-the-followers-and-following-to-the-user-model-in-the-database-so-it-is-possible-to-follow-unfollow-users-and-get-a)

## Description
This PR adds followers and following to the user's model. It also adds the required controllers and routes to be able to follow and unfollow a user.


## Testing Steps

- `git pull`
- `git checkout `
- `npm install`
- `npm run tsc`
- `npm run watch`
- Register to users
- Now you can follow an account by hitting `http://localhost:8080/api/users/follow/:followToUsername` appending the user's to follow username. Keep in mind that you must be authenticated to perform this request.
- Check that you can't follow the same account twice or follow your own account.
- You can also unfollow an account just by changing the route to `unfollow`.
- When you unfollow, check that you can't unfollow that you are not following already or unfollow your own account.



## Changes
No UI changes
